File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.size()==0) return {};
        sort(nums.begin(),nums.end());
        int n = nums.size();
        int prev = INT_MIN;
        int count = 0;
        int longest = 1;
        for (int i = 0;i<nums.size();i++){

            if (prev==nums[i]-1){
                prev = nums[i];
                count++;
            }
            else if(prev!=nums[i]) { 
                prev = nums[i];
                count = 1;
            }

            longest = max(longest, count);
            

        }
        return longest;
        
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums)==0: return 0
        arr = sorted(nums)
        longest = 1 
        count = 0
        prev = float('-inf')
        for i in range(len(arr)):
            if prev == arr[i]-1:
                count+=1 
            elif prev!=arr[i]:
                count = 1 

            prev = arr[i] 

            longest = max(longest, count)
        return longest

        # 0 1 1 2 
        

